### PR TITLE
Fix ContractRenewalCollateral

### PR DESCRIPTION
**Pull request by peterjan**
I noticed renewal failures in our logs that indicate we're not properly respecting the host's `MaxCollateral` setting. 

```json
{"level":"error","date":"2023-12-07T16:56:27Z","component":"autopilot.contractor","msg":"renewal failed, err: couldn't renew contract: RPCRenew: failed to read RPCRenewContractHostAdditions: ReadResponse: failed to validate renewal: collateral exceeds maximum: expected at most 10000000000000000000000000000 got 10000047475326975980269993984\n","hk":"ed25519:c4b451fa7ab7085763119aeb1aa34db524c5dc6416dc77c5ba82100eb9687790","fcid":"fcid:5a32d8bbfb7e6a29f874ad2ded500d46e3e9aa4675db624b7f782b7fd7f45e05"}
```

### CODE REVIEW

File: rhp/v2/contracts.go [hunk 1]
```diff
@@ -85,29 +85,27 @@ func ContractRenewalCollateral(fc types.FileContract, expectedNewStorage uint64,
...
+	// calculate the base collateral
 	baseCollateral := host.Collateral.Mul64(fc.Filesize).Mul64(extension)
+
+	// if it exceeds MaxCollateral we can't add more
 	if baseCollateral.Cmp(host.MaxCollateral) >= 0 {
 		return types.ZeroCurrency
 	}
 
-	// calculate the new collateral
-	newCollateral := host.Collateral.Mul64(expectedNewStorage).Mul64(duration)
```
Comment (ChrisSchinnerl) [inline rhp/v2/contracts.go:97]: I slightly prefer `newCollateral` since it indicates that it's not the full collateral of the renewal but the counterpart of `baseCollateral`

File: rhp/v2/contracts.go [hunk 2]
```diff
@@ -85,29 +85,27 @@ func ContractRenewalCollateral(fc types.FileContract, expectedNewStorage uint64,
...
-		if delta.Cmp(newCollateral) > 0 {
-			newCollateral = types.ZeroCurrency
-		} else {
-			newCollateral = newCollateral.Sub(delta)
-		}
+	// if the new total exceeds MaxCollateral, the renewal collateral must equal
+	// the delta to ensure we don't exceed it
+	totalCollateral := baseCollateral.Add(renewalCollateral)
+	if totalCollateral.Cmp(host.MaxCollateral) >= 0 {
+		renewalCollateral = host.MaxCollateral.Sub(baseCollateral)
```
Comment (ChrisSchinnerl) [inline rhp/v2/contracts.go:105]: I feel like this produces the same result as the existing code? I might be wrong.
Could you add a table test which hits all the edge cases and has a case for the regression to better highlight the issue?

**Review by ChrisSchinnerl — CHANGES_REQUESTED** [review]

**Comment by peterjan** [comment]
closing this out for now
